modules/core/include/opencv2/core/cuda/common.hpp

```diff
@@ -97,26 +97,6 @@ namespace cv { namespace cuda
         {
             return (total + grain - 1) / grain;
         }
-
-        template<class T> inline void bindTexture(const textureReference* tex, const PtrStepSz<T>& img)
-        {
-            cudaChannelFormatDesc desc = cudaCreateChannelDesc<T>();
-            cudaSafeCall( cudaBindTexture2D(0, tex, img.ptr(), &desc, img.cols, img.rows, img.step) );
-        }
-
-        template<class T> inline void createTextureObjectPitch2D(cudaTextureObject_t* tex, PtrStepSz<T>& img, const cudaTextureDesc& texDesc)
-        {
-            cudaResourceDesc resDesc;
-            memset(&resDesc, 0, sizeof(resDesc));
-            resDesc.resType = cudaResourceTypePitch2D;
-            resDesc.res.pitch2D.devPtr = static_cast<void*>(img.ptr());
-            resDesc.res.pitch2D.height = img.rows;
-            resDesc.res.pitch2D.width = img.cols;
-            resDesc.res.pitch2D.pitchInBytes = img.step;
-            resDesc.res.pitch2D.desc = cudaCreateChannelDesc<T>();
-
-            cudaSafeCall( cudaCreateTextureObject(tex, &resDesc, &texDesc, NULL) );
-        }
```

Comment:
It's public API and some custom modules may relay on it. Let's presume the functions. Please Add macro to check version and throw exception for CUDA 12 and above. For older CUDA versions the implementation should stay the same. Also please add "todo" to remove deprecated branch in OpenCv 5.x.